tests/test_pagination.py

```diff
@@ -13,54 +18,61 @@ def test_run(self):
-        self.select_found_catalogs(conn_id, found_catalogs, only_streams=expected_streams)
+        self.select_found_catalogs(conn_id, found_catalogs,
+                                   only_streams=expected_streams)
 
         # Run a sync job using orchestrator
-        record_count_by_stream = self.run_and_verify_sync(conn_id, expected_streams)
+        record_count_by_stream = self.run_and_verify_sync(
+            conn_id, expected_streams)
         actual_fields_by_stream = runner.examine_target_output_for_fields()
         sync_records = runner.get_records_from_target_output()
 
         for stream in expected_streams:
             with self.subTest(stream=stream):
 
-                # verify that we can paginate with all fields selected
+                # Verify that we can paginate with all fields selected
                 minimum_record_count = 50
 
                 self.assertGreater(record_count_by_stream.get(stream, -1), minimum_record_count,
-                    msg="The number of records is not over the stream max limit")
+                                   msg="The number of records is not over the stream max limit")
 
-                expected_pk = self.expected_primary_keys()
+                expected_pk = self.expected_primary_keys().get(stream, set())
                 sync_messages = sync_records.get(stream, {'messages': []}).get('messages')
 
-                # verify that the automatic fields are sent to the target
+                # Verify that the automatic fields are sent to the target
                 self.assertTrue(
                     actual_fields_by_stream.get(stream, set()).issuperset(
-                        expected_pk.get(stream, set()) |
+                        expected_pk |
                         self.expected_replication_keys().get(stream, set())),
                     msg="The fields sent to the target don't include all automatic fields"
                 )
 
-                # verify we have more fields sent to the target than just automatic fields
+                # Verify we have more fields sent to the target than just automatic fields
                 self.assertTrue(
                     actual_fields_by_stream.get(stream, set()).symmetric_difference(
-                        expected_pk.get(stream, set()) |
+                        expected_pk |
                         self.expected_replication_keys().get(stream, set())),
                     msg="The fields sent to the target don't include non-automatic fields"
                 )
 
+                # Get records
+                records = [message.get("data") for message in sync_messages
+                           if message.get('action') == 'upsert']
+                # Remove duplicate records
+                records_pks_list = [tuple(message.get(pk) for pk in expected_pk)
+                                    for message in [json.loads(t) for t in {json.dumps(d) for d in records}]]
+                records_pks_set = set(records_pks_list)
+
```

Comment:
After close analysis, IMO we should remove this test itself since we haven't done any custom pagination implementation in tap. It is handled by the Asana SDK itself on which we don't have any control, so we can't test pagination effectively using this test.